payjoin/src/receive/v2/mod.rs

```diff
@@ -60,6 +61,22 @@ pub struct Receiver {
     context: SessionContext,
 }
 
+/// A wrapper around the receiver session. The receiver session is accessible only after it has been persisted via `persist`.
+pub struct EphemeralReceiver {
+    inner: Receiver,
+}
+
+impl EphemeralReceiver {
+    pub fn persist(
+        &self,
+        persist: impl Fn(&Receiver) -> Result<(), ImplementationError>,
```

Comment:
Should probably be `Fn(key, value)` to satisfy some of #336's requests rather than Fn(Receiver).

Reply:
Ah yes! That totally slipped my mind. Thanks